linalg/sparsemat.cpp

```diff
@@ -174,9 +175,21 @@ SparseMatrix::SparseMatrix(const SparseMatrix &mat, bool copy_graph)
       }
 
       // We probably do not need to set the ownership flags here.
-      I.Reset(); I.SetHostPtrOwner(true);
-      J.Reset(); J.SetHostPtrOwner(true);
-      A.Reset(); A.SetHostPtrOwner(true);
+      if (mt != MemoryType::SIZE)
+      {
+         I.Reset(mt);
+         J.Reset(mt);
+         A.Reset(mt);
```

Comment:
The documentation of the `Reset` method says "Reset the memory and set the host memory type.", how do you know that `mt` is a host memory type?

Reply:
That documentation is a bit out-of-date, in the `Memory` class there statements like [this](https://github.com/mfem/mfem/blob/master/general/mem_manager.hpp#L697) that will get the corresponding host type even when the argument is a device type. More fundamentally I don't think this is even needed since a call to `Finalize()`, which is what will actually modify `I`, `J`, and `A`, uses `Memory::New()`, which ignores the `Memory` object's current `MemoryType`, if that sounds right to you I can revert these changes.